src/Stratis.Bitcoin.Features.Wallet/TransactionBuildOptions.cs

```diff
@@ -0,0 +1,106 @@
+        /// </summary>
+        public List<Recipient> Recipients { get; set; }
+
+        /// <summary>
+        /// Address to send change back to. Must belong to <see cref="WalletAccountReference"/>.
+        /// </summary>
+        public HdAddress ChangeAddress { get; set; }
+
+        /// <summary>
+        /// Whether to send a low, medium, or high fee. Defaults to medium.
+        /// </summary>
+        public FeeType FeeType { get; set; }
+
+        /// <summary>
+        /// Minimum number of confirmations for UTXOs to use as inputs for this transaction.
+        /// </summary>
+        public int MinConfirmations { get; set; }
+
+        /// <summary>
+        /// Data to be saved on-chain using an OP_RETURN.
+        /// </summary>
+        public string OpReturnData { get; set; }
+
+        /// <summary>
+        /// Overrides the fee rate from the <see cref="WalletFeePolicy"/> that would otherwise be used by the <see cref="WalletTransactionHandler"/>
+        /// </summary>
+        public FeeRate OverrideFeeRate { get; set; }
+
+        /// <summary>
+        /// Explicitly set the inputs that should be used for this transaction, rather than having the <see cref="WalletTransactionHandler"/> pick them itself.
+        /// </summary>
+        public List<OutPoint> SelectedInputs { get; set; }
+
+        /// <summary>
+        /// Whether to randomly order the outputs of the transaction to preserve privacy.
+        /// </summary>
+        public bool ShuffleOutputs { get; set; }
+
+        /// <summary>
+        /// Set the transaction fee explicitly for this transaction.
+        /// </summary>
+        public Money TransactionFee { get; set; }
+
+        /// <summary>
+        /// Initializes a new options object with open wallet and addresses to send to. 
+        /// </summary>
+        public TransactionBuildOptions(WalletAccountReference walletAccountReference, string password, List<Recipient> recipients)
+        {
+            Guard.NotNull(recipients, nameof(recipients));
+
+            this.WalletAccountReference = walletAccountReference;
+            this.WalletPassword = password;
+            this.Recipients = recipients;
+            InitializeDefaults();
+        }
+
+        /// <summary>
+        /// When estimating, don't need password.
+        /// </summary>
+        public TransactionBuildOptions(WalletAccountReference walletAccountReference, List<Recipient> recipients) 
+            : this(walletAccountReference, null, recipients) {}
+
+        /// <summary>
+        /// Set default values on all optional properties.
+        /// </summary>
+        private void InitializeDefaults()
+        {
+            this.ChangeAddress = null;
+            this.FeeType = FeeType.Medium;
+            this.MinConfirmations = 1;
+            this.OpReturnData = null;
+            this.OverrideFeeRate = null;
+            this.SelectedInputs = new List<OutPoint>();
+            this.ShuffleOutputs = false;
```

Comment:
We seem to be defaulting this to `true` in `WalletController.BuildTransaction`. Is it defaulted to `false` for any particular reason here? Note that shuffling by default is generally preferable as it makes finding related wallet addresses more difficult when performing chain analysis.

Reply:
It also defaulted to false in the previous implementation, with it being set true in the controller like you mentioned. 

At the moment this PR is just to recreate the same functionality, but I do agree that the defaults should be re-evaluated in the future